src/contracts/GPv2Settlement.sol

```diff
@@ -340,62 +341,28 @@ contract GPv2Settlement is ReentrancyGuard {
     }
 
     /// @dev Execute a list of arbitrary contract calls from this contract.
-    /// @param encodedInteractions The encoded list of interactions that will be
-    /// executed.
-    function executeInteractions(bytes calldata encodedInteractions) internal {
-        // Note: at every decoding step, the content of this variable is
-        // replaced with the latest decoded interaction.
-        GPv2Encoding.Interaction memory interaction;
-
-        bytes calldata remainingEncodedInteractions = encodedInteractions;
-        while (remainingEncodedInteractions.length != 0) {
-            remainingEncodedInteractions = remainingEncodedInteractions
-                .decodeInteraction(interaction);
-            executeInteraction(interaction);
-        }
-    }
-
-    /// @dev Allows settlement function to make arbitrary contract executions.
-    /// @param interaction contains address and calldata of the contract interaction.
-    function executeInteraction(GPv2Encoding.Interaction memory interaction)
+    /// @param interactions The list of interactions to execute.
+    function executeInteractions(GPv2Interaction.Data[] calldata interactions)
```

Comment:
It looks like this change removes the singular interaction execution in favor of the plural. I suppose this is because interaction execution doesn't require anything fancy anymore so its not necessary?

Sad to see it go, I was kinda pleased with my work on this.

Reply:
It just moved (I see you noticed :point_down:).